impeller/renderer/backend/metal/surface_mtl.mm

```diff
@@ -143,6 +193,27 @@
       ContextMTL::Cast(context.get())->CreateMTLCommandBuffer();
   [command_buffer commit];
   [command_buffer waitUntilScheduled];
+
+  // If there is no partial repaint, then immediately present without a blit
+  // pass.
+  if (!requires_blit_) {
+    [drawable_ present];
+    return true;
+  }
+
+  auto blit_command_buffer = context->CreateCommandBuffer();
```

Comment:
@bdero , if we are doing a final blit for partial repaint, I think I could just hop on the command buffer above and use that? I don't think it actually needs to be empty right, its just a way to ensure that everything else has been scheduled

Reply:
I tried this out a bit but ran into some memory corruption issues. not investigating, just moved the blit before the final command buffer creation for now.